x-pack/plugin/rank-rrf/src/yamlRestTest/resources/rest-api-spec/test/rrf/700_rrf_retriever_search_api_compatibility.yml

```diff
@@ -960,3 +966,55 @@ setup:
   - length: { hits.hits : 1 }
 
   - match: { hits.hits.0._id: "1" }
+
+---
+"rrf retriever with inner_hits for sub-retriever":
+
+  - do:
+      search:
+        _source: false
+        index: test
+        body:
+          retriever:
+            rrf:
+              retrievers: [
+                {
+                  standard: {
+                    query: {
+                      nested: {
+                        path: nested_inner_hits,
+                        inner_hits: {
+                          _source: false,
+                          "sort": [ {
+                              "nested_inner_hits.data": "asc"
+                            }
+                          ],
+                          fields: [ nested_inner_hits.data ]
+                        },
+                        query: {
+                          match_all: { }
+                        }
+                      }
+                    }
+                  }
+                },
+                {
+                  standard: {
+                    query: {
+                      match_all: { }
+                    }
+                  }
+                }
```

Comment:
we need to see how inner hits are scored even when when there are inner hits provided in multiple retrievers. What do we do when there are inner hits set for knn and inner hits set for a regular query? How do we unify?

Reply:
Updated the test to account for this. As things stand, we would have 2 separate entries for each retriever, one for each `inner_hit` specified (with null/empty entries if the field is not present). 
One thing to note is that if we're computing `inner_hits` for the same nested field, we have to provide custom names otherwise we'll throw an error of having duplicate inner hits.

Reply:
@pmpailis yeah, custom names is sort of expected, but rrf should be able to provide names by default, since the passed retrievers are in an array, their order is guaranteed, so, we should be able to name them automatically for folks. 


We also need to consider what it means to merge the final inner_hits view, e.g. combining the inner hits themselves via rrf for a given doc to indicate the best individual inner hit.

Reply:
> so, we should be able to name them automatically for folks.

The name is currently extracted from a `InnerHitBuilder` member in the appropriate query builders (e.g. `HasParentQueryBuilder` , `NestedQueryBuilder` , etc). Should we apply some custom logic to act iff the underlying queries are any of the above types? 

> combining the inner hits themselves via rrf for a given doc to indicate the best individual inner hit.

So give the option to  apply `rrf` to the different inner hits rankings to provide a (locally) global ranking? That would be interesting! Do you think though that this should be tracked as part of this PR or would it be something to discuss and handle separately?

Reply:
> Should we apply some custom logic to act iff the underlying queries are any of the above types?

I am saying, users forgetting to put a _name will be exceptionally common and if we can help them by automatically naming things, it would be very nice :). But this can be done in a follow up PR.

> Do you think though that this should be tracked as part of this PR or would it be something to discuss and handle separately?

Something separate for now. I would expect it to be a new inner hits option that is only available for rrf or something.